addons/html_editor/static/src/main/link/link_plugin.js

```diff
@@ -392,19 +470,36 @@ export class LinkPlugin extends Plugin {
             return linkElement;
         } else {
             // create a new link element
+            const selectedNodes = this.shared.getSelectedNodes();
+            const imageNode = selectedNodes.find((node) => node.tagName === "IMG");
+
             const link = this.document.createElement("a");
             if (!selection.isCollapsed) {
                 const content = this.shared.extractContent(selection);
                 link.append(content);
+                link.normalize();
```

Comment:
What is it ?

Reply:
It's to remove the empty node after the `append`. Sometimes the link element has an empty node child, `<a>[emptynode]abc</a>`, so sometimes we can't reset the correct selection on the image node